eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,102 @@ function GetPackageLookup($packageList) {
+  Add-Content -Path $readmePath -Value $header
+
+  # Add tables, seperate client and mgmt.
+  $readmeHeader = "# $langTitle - $moniker"
+  Add-Content -Path $readmePath -Value $readmeHeader
+  if (Test-Path (Join-Path $absolutePath -ChildPath $clientTableLink)) {
+    $clientTable = "## Client packages - $moniker`r`n"
+    $clientTable += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $clientTable
+  }
+  if (Test-Path (Join-Path $absolutePath -ChildPath $mgmtTableLink)) {
+    $mgmtTable = "## Management packages - $moniker`r`n"
+    $mgmtTable += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $mgmtTable -NoNewline
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($absolutePath, $readmeName, $serviceName, $msService)
+{
+  $readmePath = Join-Path $absolutePath -ChildPath $readmeName
+  $readmeContent = Get-Content -Path $readmePath -Raw
+  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $restContent = $Matches["content"]
+
+  $lang = $LanguageDisplayName
+  $metadataString = GenerateDocsMsMetadata -language $lang -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Set-Content -Path $readmePath -Value "$metadataString`n$restContent" -NoNewline
+}
+
+function generate-markdown-table($absolutePath, $readmeName, $packageInfo, $moniker) {
+  $content = "| Reference | Package | Source |`r`n|---|---|---|`r`n" 
+  # Here is the table, the versioned value will
+  foreach ($pkg in $packageInfo) {
+    if (!$pkg.VersionGA -and "latest" -eq $moniker) {
+      continue
+    }
+    if (!$pkg.VersionPreview -and "preview" -eq $moniker) {
+      continue
+    }
+    $repositoryLink = $RepositoryUri
+    $packageLevelReadme = &$GetPackageLevelReadmeFn -packageMetadata $pkg
+    $referenceLink = "[$($pkg.DisplayName)]($packageLevelReadme-readme.md)"
+    if (!(Test-Path (Join-Path $absolutePath -ChildPath "$packageLevelReadme-readme.md"))) {
+      $referenceLink = $pkg.DisplayName
+    }
+    $githubLink = $GithubUri
+    if ($pkg.PSObject.Members.Name -contains "FileMetadata") {
+      $githubLink = "$GithubUri/blob/main/$($pkg.FileMetadata.DirectoryPath)"
+    }
+    $line = "|$referenceLink|[$($pkg.Package)]($repositoryLink/$($pkg.Package))|[Github]($githubLink)|`r`n"
+    $content += $line
+  }
+  Set-Content -Path (Join-Path $absolutePath -ChildPath $readmeName) -Value $content -NoNewline
+}
+
+function generate-service-level-readme($readmeBaseName, $pathPrefix, $packageInfos, $serviceName) {
+  # Add ability to override
+  # Fetch the service readme name
+  $monikers = @("latest", "preview")
+  $msService = GetDocsMsService -packageInfo $packageInfos[0] -serviceName $serviceName
+  foreach($moniker in $monikers) {
+    $absolutePath = "$DocRepoLocation/$pathPrefix/$moniker/"
```

Comment:
This should be something like `$readmeFolder` instead of `$absolutePath` (same with the variable name in the function definitions). `$DocRepoLocation` could also potentially be a relative path which would make `$absolutePath` relative.

The name "absolute path" carries with it the connotations that it's the final path a function would work on. However, in all usages of `$absolutePath` it is being combined with another value to produce another final path.